tests/test_cases/test_cocotb/test_deprecated.py

```diff
@@ -0,0 +1,19 @@
+import cocotb
+import warnings
+
+
+@cocotb.test()
+async def test_returnvalue_deprecated(dut):
+    @cocotb.coroutine
+    def get_value():
+        yield cocotb.triggers.Timer(1, units='ns')
+        raise cocotb.result.ReturnValue(42)
+
+    with warnings.catch_warnings(record=True) as w:
+        # Cause all warnings to always be triggered.
+        warnings.simplefilter("always")
+        val = await get_value()
+    assert val == 42
+    assert len(w) == 1
+    assert issubclass(w[-1].category, DeprecationWarning)
+    assert "use return instead" in str(w[-1].message)
```

Comment:
Any reason for the `str`?
```suggestion
    assert "use return instead" in w[-1].message
```

Reply:
I was just following the [example](https://docs.python.org/3.8/library/warnings.html#testing-warnings). Will remove.

Reply:
Huh, interesting. Fine either way then, I guess.

Reply:
Turns out it was there for a very good reason, `.message` isn't a string...

Reply:
Whoops, `.message` is the `Warning` instance.